File: communication_bk/src/main/java/net/tomofiles/skysign/communication/domain/communication/UploadMission.java
package net.tomofiles.skysign.communication.domain.communication;

import lombok.AccessLevel;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.ToString;

@Getter
@RequiredArgsConstructor(access = AccessLevel.PACKAGE)
@EqualsAndHashCode(of = {"id"})
@ToString
class UploadMission {
    private final CommandId id;
    private final MissionId missionId;

    public static UploadMission empty(CommandId id) {
        return new UploadMission(id, null);
    }
}

File: remote-communication/pkg/communication/domain/communication/upload_mission.go
package communication

// MissionID .
type MissionID string

// UploadMission .
type UploadMission struct {
	commandID CommandID
	missionID MissionID
}

// NewUploadMission .
func NewUploadMission(commandID CommandID, missionID MissionID) *UploadMission {
	return nil
}
